apps/coscrad-frontend/src/components/resources/digital-text/digital-text-detail.full-view.presenter.tsx

```diff
@@ -1,19 +1,151 @@
-            {pages.length > 0 ? <PagesPresenter pages={pages} /> : null}
+            {pages.length > 0 ? (
+                // TODO Offer multiple views here
+                <PagesPresenter
+                    pages={pages}
+                    currentPageIdentifier={selectedPageIdentifier}
+                    onSubmitNewContent={({ text, languageCode, pageIdentifier }) =>
+                        dispatch(
+                            executeCommand({
+                                type: 'ADD_CONTENT_TO_DIGITAL_TEXT_PAGE',
+                                payload: {
+                                    aggregateCompositeIdentifier: {
+                                        type: AggregateType.digitalText,
+                                        id,
+                                    },
+                                    text,
+                                    languageCode,
+                                    pageIdentifier,
+                                },
+                            })
+                        )
+                    }
+                />
+            ) : null}
+            <Button
+                onClick={() => {
+                    setCurrentIndex(cyclicDecrement(currentIndex, pages.length));
+                }}
+            >
+                {'<< PREV'}
+            </Button>{' '}
+            <Button
+                onClick={() => {
+                    setCurrentIndex(cyclicIncrement(currentIndex, pages.length));
+                }}
+            >
+                {'NEXT >>'}
+            </Button>
+            {/* TODO I am a back-end developer */}
+            <br />
+            <NewPageForm
+                existingPageIdentifiers={allExistingPageIdentifiers}
+                onSubmitPageIdentifier={(pageIdentifier) => {
+                    dispatch(
+                        executeCommand({
+                            type: 'ADD_PAGE_TO_DIGITAL_TEXT',
+                            payload: {
+                                aggregateCompositeIdentifier,
+                                identifier: pageIdentifier,
+                            },
+                        })
+                    );
+                }}
+            />
+            {!isPublished ? (
+                <PublicationForm
+                    onSubmitForPublication={() => {
+                        dispatch(
+                            executeCommand({
+                                type: 'PUBLISH_RESOURCE',
+                                payload: {
+                                    aggregateCompositeIdentifier: aggregateCompositeIdentifier,
+                                },
+                            })
+                        );
+                    }}
+                />
+            ) : (
+                <Typography variant="body2">Published</Typography>
+            )}
+            <br />
+            {/* TODO Include tags */}
```

Comment:
Fix the tag repository to persist events and confirm that they show up in the query flow tests.

Reply:
[Here](https://www.pivotaltracker.com/story/show/186539279) is the story.